data/completion/complete.sh

```diff
@@ -0,0 +1,73 @@
+# -*- bash -*-
+
+# _complete_from_snap serialises the tab completion request and sends it off to
+# the appropriate 'snap run --command=complete', and de-serialises the response
+# into the usual tab completion result.
```

Comment:
Perhaps an overview of the flow of control would be good here (or at the top of the file) as well. Eg:
```
# How snap command completion works is:
# 1. snappy's complete.sh is sourced into the user's shell environment
# 2. user performs '<command> <tab>'. If '<command>' is a snap command,
#    proceed to step '3', otherwise perform normal bash completion
# 3. run 'snap run --command=complete ...', converting bash completion
#    environment into serialized command line arguments
# 4. 'snap run --command=complete ...' exec()s 'etelpmoc.sh' within the snap's
#    runtime environment and confinement
# 5. 'etelpmoc.sh' takes the serialized command line arguments from step '3'
#    and puts them back into the bash completion environment variables
# 6. 'etelpmoc.sh' sources the snap's 'completer' script, performs the bash
#    completion and serializes the resulting completion environment variables
#    by printing to stdout the results in a format that snappy's complete.sh
#    will understand, then exits
# 7. control returns to snappy's 'complete.sh' and it deserializes the output
#    from 'etelpmoc.sh', validates the results and puts the validated results
#    into the bash completion environment variables
# 8. bash displays the results to the user
```